resources/mediaplayer/src/main/java/org/mobicents/media/server/impl/resource/mediaplayer/audio/wav/WavTrackImpl.java

```diff
@@ -220,22 +255,29 @@ private void getFormat(InputStream stream) throws IOException {
      * @throws java.io.IOException
      */
     private int readPacket(byte[] packet, int offset, int psize) throws IOException {
-        int length = 0;
-        try {
-            while (length < psize) {
-                int len = inStream.read(packet, offset + length, psize - length);
-                if (len == -1) {
-                    return length;
-                }
-                length += len;
-            }
-            return length;
-        } catch (Exception e) {
-            logger.error(e);
+     
+        int bytesRead=0;
+        
+        if (psize == 0) { 
+            return 0; 
+        } 
+        
+        if (buff.capacity()< psize){
```

Comment:
why dont we use a predefined buffer size, and in case the incoming packet is bigger, deny traffic?? redimensioning the buffer for any packet size seems a bit unsafe...